pipeline/test_executor.groovy

```diff
@@ -0,0 +1,173 @@
+def nodeName = "centos-7"
+def sharedLib
+def retVal
+
+def getCLI(){
+	/*
+		Generates the CLI using the arguments provided and returns it.
+	*/
+	def cephVersion = 'pacific'
+    if("3." in "${params.RHCS_Build}"){
+        cephVersion = 'luminous'
+    }
+    else if("4." in "${params.RHCS_Build}"){
+        cephVersion = 'nautilus'
+    }
+
+	def cli = "--rhbuild ${params.RHCS_Build}"
+	cli += " --platform ${params.Platform}"
+	cli += " --build ${params.Build}"
+	cli += " --v2"
+	cli += " --global-conf conf/${cephVersion}/${params.Group}/${params.Conf}.yaml"
+	cli += " --inventory conf/inventory/${params.Inventory}.yaml"
+	cli += " --suite suites/${cephVersion}/${params.Group}/${params.Suite}.yaml"
+	cli += " ${params.Additional_Args}"
+
+	if(params.Override_Ceph_Base_URL?.trim()){
+		cli += " --rhs-ceph-repo ${params.Override_Ceph_Base_URL}"
+	}
+	if(params.Additional_Repo_File?.trim()){
+		cli += " --add-repo ${params.Additional_Repo_File}"
+	}
+	if(params.Hotfix_Repo_File?.trim()){
+		cli += " --hotfix-repo ${params.Hotfix_Repo_File}"
+	}
+	if(params.Override_Container_Image?.trim()){
+		container = params.Override_Container_Image.split(".com/")
+		docker_registry = container[0] + ".com"
+		docker = container[1].split(":")
+		docker_tag = docker.last()
+		docker_image = docker.first()
+		cli += " --insecure-registry"
+		cli += " --docker-registry " + docker_registry
+		cli += " --docker-tag " + docker_tag
+		cli += " --docker-image " + docker_image
+	}
+	return cli
+}
+
+def buildArtifactDetails(sharedLib){
```

Comment:
As `sharedLib` is defined globally, you can use it directly within the method.

Reply:
As discussed, going on with passing it as argument itself.